cluster/gce/coreos/master.yaml

```diff
@@ -0,0 +1,169 @@
+#cloud-config
+
+coreos:
+  units:
+    - name: kubernetes-node-params.service
+      command: start
+      content: |
+        [Unit]
+        Description=Kubernetes Configuration Fetcher
+        Requires=network-online.target
+        After=network-online.target
+
+        [Service]
+        Type=oneshot
+        RemainAfterExit=yes
+        ExecStart=/bin/curl --fail --silent --show-error \
+        -H "X-Google-Metadata-Request: True" \
+        -o /etc/kubernetes-node-params \
+        http://169.254.169.254/computeMetadata/v1/instance/attributes/kubernetes-node-params
+
+    - name: etcd.service
+      command: start
+      content: |
+        [Unit]
+        Description=etcd
+        Requires=kubernetes-node-params.service
+        After=kubernetes-node-params.service
+
+        [Service]
+        EnvironmentFile=/etc/kubernetes-node-params
+        PermissionsStartOnly=true
+        ExecStartPre=/usr/bin/mkdir -p /opt/kubernetes/.kube
+        ExecStartPre=/usr/bin/mv /etc/${MASTER_NAME}.tar.gz /opt/kubernetes/.kube
+        ExecStartPre=/usr/bin/tar xzf /opt/kubernetes/.kube/${MASTER_NAME}.tar.gz -C /opt/kubernetes/.kube --overwrite
+        ExecStart=/usr/bin/etcd \
+        --name ${MASTER_NAME} \
+        --discovery ${ETCD_DISCOVERY} \
+        --addr $private_ipv4:4001 \
+        --bind-addr 0.0.0.0 \
+        --data-dir /var/lib/etcd \
+        --peer-addr $private_ipv4:2380 \
+        --peer-bind-addr $private_ipv4:2380 \
+        --peer-ca-file=/opt/kubernetes/.kube/ca.crt \
+        --peer-cert-file=/opt/kubernetes/.kube/issued/${MASTER_NAME}.crt \
+        --peer-key-file=/opt/kubernetes/.kube/private/${MASTER_NAME}.key
+        Restart=always
+        RestartSec=10s
+
+    - name: fleet.socket
+      command: start
+      content: |
+        [Socket]
+        ListenStream=/var/run/fleet.sock
+
+    - name: fleet.service
+      command: start
+      content: |
+        [Unit]
+        Description=fleet daemon
+        Wants=etcd.service
+        After=etcd.service
+        Wants=fleet.socket
+        After=fleet.socket
+        [Service]
+        EnvironmentFile=/etc/kubernetes-node-params
+        ExecStart=/usr/bin/fleetd
+        Restart=always
+        RestartSec=10s
+
+# TODO: Organize these into targets.
+    - name: kubernetes-install-master.service
+      command: start
+      content: |
+        [Unit]
+        Description=Kubernetes Installer Scripts
+        Requires=network-online.target
+        After=network-online.target
+        Requires=kubernetes-node-params.service
+        After=kubernetes-node-params.service
+
+        [Service]
+        Type=oneshot
+        RemainAfterExit=yes
+        EnvironmentFile=/etc/kubernetes-node-params
+        ExecStartPre=/usr/bin/mkdir -p /opt/kubernetes/pkg
+        ExecStartPre=/usr/bin/curl --location --create-dirs --output /opt/kubernetes/pkg/kubernetes-server-linux-amd64.tar.gz ${SERVER_BINARY_TAR_URL}
```

Comment:
Borrowing from `download-or-bust`, I would add something like `--connect-timeout 20 --retry 6 --retry-delay 10` to any of the `curl` commands. It's really bad if we can't get the software.